src/lib/es5.d.ts

```diff
@@ -1570,7 +1566,7 @@ interface Int8Array {
       * @param thisArg An object to which the this keyword can refer in the callbackfn function.
       * If thisArg is omitted, undefined is used as the this value.
       */
-    filter(callbackfn: (value: number, index: number, array: Int8Array) => any, thisArg?: any): Int8Array;
+    filter(callbackfn: (value: number, index: number, array: this) => any, thisArg?: any): T;

```

Comment:
Rather than return `T`, either return `this` or return `TypedArray` with a subtyped overload in each specific typed-array constructor (e.g. return `Int32Array` in the `Int32Array` interface).